cmd/torrent-create/main.go

```diff
@@ -26,6 +26,9 @@ func init() {
 func main() {
 	b := torrent.Builder{}
 	for _, filename := range flag.Args() {
+		if _, err := os.Stat(filename); os.IsNotExist(err) {
+			log.Fatal(err)
+		}
 		if err := filepath.Walk(filename, func(path string, info os.FileInfo, err error) error {
```

Comment:
I think you've introduced a race via LBYL. You should check the error inside the Walk? This is the reason for the panic.